bot/exts/utilities/bookmark.py

```diff
@@ -23,6 +23,40 @@
 )
 
 
+class BookmarkTitleSelectionForm(discord.ui.Modal):
+    """
+    The form where a user can fill in a custom title for their bookmark & submit it.
+
+    This form is only available when the command is invoked from a context menu.
+    """
+
+    def __init__(
+            self,
+            message: discord.Message,
+            action_bookmark_function: Callable[[discord.TextChannel, discord.Member, discord.Message, str], None],
+    ):
+        super().__init__(timeout=1000, title="Name your bookmark")
+        self.message = message
+        self.action_bookmark = action_bookmark_function
+
+    bookmark_title = discord.ui.TextInput(
+        label="Choose a title for you bookmark",
+        placeholder="Type your bookmark title here",
+        default="Bookmark",
+        max_length=50,
+        min_length=0,
+        required=False
+    )
```

Comment:
As a rule of thumb we want the `bookmark_title` to come before `__init__`
```suggestion
    bookmark_title = discord.ui.TextInput(
        label="Choose a title for you bookmark",
        placeholder="Type your bookmark title here",
        default="Bookmark",
        max_length=50,
        min_length=0,
        required=False
    )

    def __init__(
            self,
            message: discord.Message,
            action_bookmark_function: Callable[[discord.TextChannel, discord.Member, discord.Message, str], None],
    ):
        super().__init__(timeout=1000, title="Name your bookmark")
        self.message = message
        self.action_bookmark = action_bookmark_function
```